src/ir.js

```diff
@@ -203,7 +205,8 @@ exports.createIR = (query) => {
                 let save = currentGroupPath
                 let deps = []
                 for (let d in rhsdeps) if (isVar(d) && !(d in keydeps)) deps.push(d)
-                let plus = deps.map(ident)
+                // plus may contain existing keys in currentGroupPath
+                let plus = filterKeysFromGroupPath(deps.map(ident))
```

Comment:
Could the same be achieved by adding everything from currentGroupPath to keydeps before or after line 198?

Reply:
Turns out adding everything from currentGroupPath to keydeps may not work. since deps = rhsdeps - keydeps, which is also used to create temp var: createFreshTempVar(deps) in line 211. If we remove duplicate keys from deps, the created temporary will be incorrect.